ra2ce/runners/analysis_runner_factory.py

```diff
@@ -33,56 +31,58 @@
 
 class AnalysisRunnerFactory:
     @staticmethod
-    def get_supported_runners(ra2ce_input: ConfigWrapper) -> list[Type[AnalysisRunner]]:
+    def get_supported_runners(
+        analysis_collection: AnalysisCollection,
+    ) -> list[AnalysisRunner]:
         """
         Gets the supported analysis runners for the given input.
 
         Args:
-            ra2ce_input (Ra2ceInput): Input representing a set of network and analysis ini configurations.
+            analysis_collection (AnalysisCollection): Collection of analyses to run.
 
         Returns:
             list[AnalysisRunner]: Supported runners for the given configuration.
         """
-        if not ra2ce_input.analysis_config:
+        if not analysis_collection:
             return []
-        _available_runners = [
-            DamagesAnalysisRunner,
-            LossesAnalysisRunner,
-            AdaptationAnalysisRunner,
-        ]
-        _supported_runners = [
-            _runner for _runner in _available_runners if _runner.can_run(ra2ce_input)
+        _available_runners: list[AnalysisRunner] = [
+            DamagesAnalysisRunner(),
+            LossesAnalysisRunner(),
+            AdaptationAnalysisRunner(),
         ]
-        if not _supported_runners:
-            _err_mssg = "No analysis runner found for the given configuration."
-            logging.error(_err_mssg)
-            raise ValueError(_err_mssg)
+        _supported_runners = []
+        for _analysis in analysis_collection.analyses:
+            _supported_runner = [
+                _runner
+                for _runner in _available_runners
+                if _runner.can_run(_analysis, analysis_collection)
+            ]
```

Comment:
given the logic of only getting the first supported runner, it might be best to use the `next((iterator), None)` expression instead, this way we know we get `None` or just a single runner.